services/src/main/java/org/apache/druid/cli/CliCoordinator.java

```diff
@@ -192,6 +194,12 @@ protected List<? extends Module> getModules()
     modules.add(JettyHttpClientModule.global());
 
     if (isSegmentMetadataCacheEnabled) {
+      if (!properties.getOrDefault(SERVERVIEW_TYPE_PROPERTY, "http").equals("http")) {
+        throw new RuntimeException(
+            "CentralizedDatasourceSchema feature is incompatible with Zookeeper based segment discovery. "
```

Comment:
I think we should not mention zookeeper but just the value of `properties.getOrDefault(SERVERVIEW_TYPE_PROPERTY)`. If tomorrow we add new eay of announcing serviewViews that we need not change this piece of code.